pkg/rtc/participant.go

```diff
@@ -395,22 +403,21 @@ func (p *ParticipantImpl) SetPermission(permission *livekit.ParticipantPermissio
 }
 
 func (p *ParticipantImpl) ToProto() *livekit.ParticipantInfo {
-	grants := p.ClaimGrants()
+	p.lock.RLock()
 	info := &livekit.ParticipantInfo{
 		Sid:         string(p.params.SID),
 		Identity:    string(p.params.Identity),
 		Name:        string(p.params.Name),
 		State:       p.State(),
 		JoinedAt:    p.ConnectedAt().Unix(),
 		Version:     p.version.Inc(),
-		Permission:  grants.Video.ToPermission(),
+		Permission:  p.grants.Video.ToPermission(),
+		Metadata:    p.grants.Metadata,
 		Region:      p.params.Region,
 		IsPublisher: p.isPublisher.Load(),
 	}
+	p.lock.RUnlock()
 	info.Tracks = p.UpTrackManager.ToProto()
-	if p.params.Grants != nil {
-		info.Metadata = grants.Metadata
-	}
 
```

Comment:
Made this change to take the value `grants` field. Not sure why this was taking from `Params`. Guess it is okay as `Params.Grants` is a pointer and it was stored as is and `Metadata` is a string and even if re-assigned to a new metadata value, it would be reflected in the underlying structure.

Reply:
it really shouldn't have been looking at params.Grants. It could not be nil either. I think the code was initially written when I first migrated permissions into grants, and was thinking about the initial rollout.